src/contracts/yas_pool.cairo

```diff
@@ -47,14 +58,102 @@ mod YASPool {
         tick: i32
     }
 
+    #[derive(Drop, starknet::Event)]
+    struct SwapExecuted {
+        sender: ContractAddress,
+        recipient: ContractAddress,
+        amount_0: i256,
+        amount_1: i256,
+        sqrt_price_X96: FixedType,
+        liquidity: u128,
+        tick: i32
+    }
+
+    #[derive(Copy, Drop, Serde, starknet::Store)]
+    struct Slot0 {
+        // the current price
+        sqrt_price_X96: FixedType,
+        // the current tick
+        tick: i32,
+        // the current protocol fee as a percentage of the swap fee taken on withdrawal
+        // represented as an integer denominator (1/x)%
+        fee_protocol: u8,
+    }
+
+    #[derive(Copy, Drop)]
+    struct SwapCache {
+        // the protocol fee for the input token
+        fee_protocol: u8,
+        // liquidity at the beginning of the swap
+        liquidity_start: u128,
+        // the timestamp of the current block
+        block_timestamp: u64,
+        // the current value of the tick accumulator, computed only if we cross an initialized tick
+        tick_cumulative: i64,
+        // the current value of seconds per liquidity accumulator, computed only if we cross an initialized tick
+        seconds_per_liquidity_cumulative_X128: u256,
+        // whether we've computed and cached the above two accumulators
+        computed_latest_observation: bool
```

Comment:
Isn't this only used for the Oracle?

Reply:
✅

Reply:
I removed these fields since they are only used by the oracle